### DISCUSSION TITLE
I want to create myself odds data using .xlsx

### DISCUSSION THREAD

**Discussion by guanjen375** [kyleskom/NBA-Machine-Learning-Sports-Betting#160] [General]
I want to create my .sqlite file using the .xlsx provided by yourself.

Here is my code:

import random
import time
import pandas as pd
import sqlite3
import os

from datetime import datetime
from tqdm import tqdm

df_data = []

path = '../../Odds-Data/Odds-Data-Clean/'
for filename in os.listdir(path):
    if not filename.startswith('~') and filename.endswith('.xlsx'):
        data = pd.read_excel(path+filename)
        for row in data.itertuples():
            print(data.columns)
            day = row[2]
            home_team = row[3]
            away_team = row[4]
            ou = row[5]
            if(int(row[7])<0):
                spread = int(row[6])
            else:
                spread = int(row[6])*(-1)
            ML_HOME = row[7]
            ML_AWAY = row[8]
            point = row[9]
            diff = row[10]
            print(day)

Unfortunately, I got fail. The error message is as follow: 

<img width="1447" alt="image" src="https://user-images.githubusercontent.com/40448600/227766138-8ceb15c4-319e-45f3-ba18-2aedd733eb70.png">

I have checked my .xlsx and could not find the error:

<img width="768" alt="image" src="https://user-images.githubusercontent.com/40448600/227766176-99fc7f2e-6214-45c4-9293-98a2c9e862c8.png">

Do you have any comment on how to solve this problem?

**Comment by nealmick** [comment]
Maybe this solution works, i think the error is little cutoff? maybe is there more error message below the screenshot?
``` 
df = pd.read_excel(path+filename)
f = df.to_dict()
for x in range(len(f['Spread'])):  
                home = f['Home'][x]
                visitor = f['Away'][x]
                ml = f['ML_Home'][x]
                spread = f['Spread'][x]
                date = f['Date'][x]


```

**Reply by guanjen375** [reply]
> Maybe this solution works, i think the error is little cutoff? maybe is there more error message below the screenshot?
> 
> ```
> df = pd.read_excel(path+filename)
> f = df.to_dict()
> for x in range(len(f['Spread'])):  
>                 home = f['Home'][x]
>                 visitor = f['Away'][x]
>                 ml = f['ML_Home'][x]
>                 spread = f['Spread'][x]
>                 date = f['Date'][x]
> ```

The error message is as follows:

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/40448600/227913839-ee406ebe-07cc-4f9c-8286-51fd4d3a2711.png">

**Reply by nealmick** [reply]
This means there is no date. Check xlsx file for date. Maybe try an older season.

**Reply by guanjen375** [reply]
Now I can Create my own .sqlite from the directory /NBA-Machine-Learning-Sports-Betting/Odds-Data/Odds-Data-Clean/* by delete 2022-23.xlsx. My next step is to download the teams data using Get_Data.py from 2007 to 2022 (not contains 2023) and use Create_Games.py to create data.sqlite. If there is another question then I will ask you for help. Thank you.
